### PR TITLE
Av1 improve

### PR DESCRIPTION
**Pull request by feiwan1**

### CODE REVIEW

**Comment by feiwan1** [comment]
Did av1 full test locally in TGL, shows no regression.

File: libavcodec/cbs_av1_syntax_template.c
Comment (xuguangxin) [inline libavcodec/cbs_av1_syntax_template.c:358]: could we have some comment, why we need change this?

File: libavcodec/av1dec.c
Comment (xuguangxin) [inline libavcodec/av1dec.c:569]: maybe a better log is "add file grain support to vaapi"
will this impact dxva?

File: libavcodec/av1dec.h
Comment (xuguangxin) [inline libavcodec/av1dec.h]: indent

File: libavcodec/av1dec.c
Comment (xuguangxin) [inline libavcodec/av1dec.c:943]: will this impact dxva?

File: libavcodec/vaapi_decode.c
Comment (xuguangxin) [inline libavcodec/vaapi_decode.c:580]: will this impact dxva?
why it will leak surfaces?
It's better to move this commit earlier than the previous one, else film grain clips will have an issue with https://github.com/intel-media-ci/ffmpeg/pull/412/commits/db2cf91690316c44cf16e93931979a9f9e0d18a3

File: libavcodec/vaapi_av1.c
Comment (xuguangxin) [inline libavcodec/vaapi_av1.c:66]: this can be split to multiple commits
each - is a commit

**Comment by xuguangxin** [comment]
@galinart could you help review and try this on windows?
it will impact av1 dxva too.

thanks

**Comment by xuguangxin** [comment]
@uartie , any way to add @galinart to collaborator so we can add him to reviewers.

also, is it possible to disable the "rebase and merge" button for everyone?
we never want merge commits for this repo

thanks

**Comment by uartie** [comment]
> @uartie , any way to add @galinart to collaborator so we can add him to reviewers.
> 

Yes, I will add @galinart

> also, is it possible to disable the "rebase and merge" button for everyone?
> we never want merge commits for this repo
> 

No.  GitHub requires at least one option is enabled ☹️ .  We just need to avoid using it.

> thanks

**Comment by uartie** [comment]
> > @uartie , any way to add @galinart to collaborator so we can add him to reviewers.
> 
> Yes, I will add @galinart
> 

Collaborator invitation sent to @galinart

**Comment by uartie** [comment]
> > also, is it possible to disable the "rebase and merge" button for everyone?
> > we never want merge commits for this repo
> 
> No. GitHub requires at least one option is enabled ☹️ . We just need to avoid using it.
> 

I have enabled "Require pull request reviews before merging"... but max GitHub setting is 6 required reviews.  Also, I enabled "Require conversation resolution before merging".  This should at least keep the "Rebase and Merge" button disabled for most cases.

**Comment by feiwan1** [comment]
@xuguangxin  @galinart, all my change doesn't impact DXVA2/NVDEC,tf_display only used in VAAPI case. But still have question on how DXVA2/NVDEC film grain works? DXVA2/NVDEC does't have interface to support 2 output surface like what VA-API does for each film grain frame. So I guess the only output surface will be applied with film grain. Then according current logic, the output will be used as reference of next frame. But I got from some doc that only reconstruct frame without apply film grain can be used for reference.

File: libavcodec/cbs_av1_syntax_template.c
Comment (feiwan1) [reply libavcodec/cbs_av1_syntax_template.c:358]: OK, will add it.

File: libavcodec/av1dec.c
Comment (feiwan1) [reply libavcodec/av1dec.c:569]: This is common part for all av1 hw dec, and the change will happen on some basic struct. So I just want to avoid to mention "vaapi only thing". Anyway, I will think about it, to let the message more acceptable to community.

File: libavcodec/av1dec.h
Comment (feiwan1) [reply libavcodec/av1dec.h]: Will modify, thanks.

File: libavcodec/av1dec.c
Comment (feiwan1) [reply libavcodec/av1dec.c:943]: No impact, just for vaapi.

File: libavcodec/vaapi_decode.c
Comment (feiwan1) [reply libavcodec/vaapi_decode.c:580]: Will modify, thanks.

File: libavcodec/vaapi_av1.c
Comment (feiwan1) [reply libavcodec/vaapi_av1.c:66]: OK, I will check it. thanks.

**Comment by xhaihao** [comment]
> @xuguangxin @galinart, all my change doesn't impact DXVA2/NVDEC,tf_display only used in VAAPI case. But still have question on how DXVA2/NVDEC film grain works? DXVA2/NVDEC does't have interface to support 2 output surface like what VA-API does for each film grain frame. So I guess the only output surface will be applied with film grain. Then according current logic, the output will be used as reference of next frame. But I got from some doc that only reconstruct frame without apply film grain can be used for reference.

Right, the reconstruct is used as reference, the film grain output is for rendering.

**Comment by feiwan1** [comment]
HI @xuguangxin  @galinart  any further comments? I am ready to submit this patch to community.

**Review by xuguangxin — APPROVED** [review]

**Comment by feiwan1** [comment]
Hi @a1rwulf , I have submit this patch set to community, could you help comment on it in ffmpeg mail list? Thanks.
https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=4197

**Comment by feiwan1** [comment]
Submitted this PR to ffmpeg community https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=4197, and wait for review. Close this one.

**Comment by a1rwulf** [comment]
@feiwan1 thx for the fixes and for submitting it to the ML.
I'll try to see if/how I can help.

**Comment by rmkimathi** [comment]
> @feiwan1 thx for the fixes and for submitting it to the ML.
> I'll try to see if/how I can help.

Also waiting for the merge to save me from patching after every git pull ....

**Pull request closed without merging**
